src/Microsoft.SqlTools.ServiceLayer/ShowPlan/ShowPlanGraph/XmlPlanNodeBuilder.cs

```diff
@@ -371,6 +374,100 @@ private IEnumerable<BaseStmtInfoType> EnumStatements(StmtBlockType statementBloc
             }
         }
 
+        private Description ParseDescription(ShowPlanGraph graph, string XmlDocument, int index)
+        {
+
+            Description description = new Description();
+            description.HasMissingIndex = true;
+            description.QueryText = graph.Statement;
+
+            XmlDocument stmtXmlDocument = new XmlDocument();
+            stmtXmlDocument.LoadXml(XmlDocument);
+            var nsmgr = new XmlNamespaceManager(stmtXmlDocument.NameTable);
+            nsmgr.AddNamespace("shp", "http://schemas.microsoft.com/sqlserver/2004/07/showplan");
+            XmlNode rootNode = stmtXmlDocument.DocumentElement;
+
+            XmlNode missingIndexes = rootNode.SelectSingleNode("descendant::shp:MissingIndexes", nsmgr);
+            //Returning null if no missing indexes are found in the xml file
+            if (missingIndexes == null)
+            {
+                return null;
+            }
+            XmlNodeList indexGroups = missingIndexes.SelectNodes("descendant::shp:MissingIndexGroup", nsmgr);
+
+            bool memoryOptimzed = false;
+            XmlNode scan = rootNode.SelectSingleNode("descendant::shp:IndexScan", nsmgr);
+            if (scan == null)
+            {
+                scan = rootNode.SelectSingleNode("descendant::shp:TableScan", nsmgr);
+            }
+            if (scan != null && scan.Attributes["Storage"] != null)
+            {
+                if (0 == string.Compare(scan.Attributes["Storage"].Value, "MemoryOptimized", StringComparison.Ordinal))
+                {
+                    memoryOptimzed = true;
+                }
+            }
+
+            const string createIndexTemplate = "CREATE NONCLUSTERED INDEX [<Name of Missing Index, sysname,>]\r\nON {0}.{1} ({2})\r\n";
```

Comment:
quote/escape schema/table

Reply:
I got this code from ssms. It seems that names are already escaped in the query plan. 
![image](https://user-images.githubusercontent.com/6816294/151445934-b5123399-89c1-4a65-9ff2-ce69c0defff4.png)

Reply:
Oh huh, that's interesting. I would definitely recommend adding comments indicating that.

Could you check with a database/schema/table that have ] in their names to verify that things are being escaped correctly?

Reply:
![image](https://user-images.githubusercontent.com/6816294/151452897-84c57a19-aab2-4750-9258-e3bf7ca6c417.png)
It seems those names are properly escaped. The original table name for the query was [master].[dbo].[insert\\]l]